src/libcore/fmt/mod.rs

```diff
@@ -1612,5 +1622,14 @@ impl<'b, T: ?Sized + Debug> Debug for RefMut<'b, T> {
     }
 }
 
+#[stable(feature = "rust1", since = "1.9.0")]
+impl<T: ?Sized + Debug> Debug for UnsafeCell<T> {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        f.debug_struct("UnsafeCell")
+            .field("value", &unsafe { &*self.get() })
```

Comment:
I think it's fine to implement Debug, but it can't show any fields. Printing the ptr value is pointless, since the pointer is just to the interior of the value.